File: solutions/python/difference-of-squares/1/difference_of_squares.py
def square_of_sum(number):

    return (sum(range(1, number + 1))) ** 2


def sum_of_squares(number):
    n=number
    return (n*(n+1)*(2*n+1))/6


def difference_of_squares(number):
    return square_of_sum(number) - sum_of_squares(number)

File: solutions/javascript/difference-of-squares/1/difference-of-squares.js
//
// This is only a SKELETON file for the 'Difference Of Squares' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export class Squares {
  constructor(n) {
    this.n=n;
  }

  get sumOfSquares() {
     const n = this.n;
    return (n*(n+1)*(2*n+1))/6;
  }

  get squareOfSum() {
    const n = this.n;

  const sum = (n * (n + 1)) / 2;
  return sum * sum;
  }

  get difference() {
    return this.squareOfSum - this.sumOfSquares;
  }
}
